resource_processor/shared/config.py

```diff
@@ -30,19 +31,51 @@ def get_config(logger_adapter) -> dict:
     config["arm_client_id"] = os.environ["ARM_CLIENT_ID"]
     config["arm_tenant_id"] = os.environ["AZURE_TENANT_ID"]
 
-    # Only set client secret if MSI is disabled
-    config["arm_client_secret"] = os.environ["ARM_CLIENT_SECRET"] if config["arm_use_msi"] == "false" else ""
+    if config["arm_use_msi"] == "false":
+        # These are needed when running locally
+        config["arm_client_secret"] = os.environ["ARM_CLIENT_SECRET"]
+        config["aad_tenant_id"] = os.environ["AAD_TENANT_ID"]
+        config["application_admin_client_id"] = os.environ["APPLICATION_ADMIN_CLIENT_ID"]
+        config["application_admin_client_secret"] = os.environ["APPLICATION_ADMIN_CLIENT_SECRET"]
+
+        # TODO: remove this
+        config["enable_local_debugging"] = "true"
+
+    # TODO: try to remove this (test on Azure)
+    else:
+        config["arm_client_secret"] = ""
+        config["aad_tenant_id"] = ""
+        config["application_admin_client_id"] = ""
+        config["application_admin_client_secret"] = ""
 
     # Create env dict for porter
     config["porter_env"] = {
         "HOME": os.environ["HOME"],
         "PATH": os.environ["PATH"],
+        "KEY_VAULT_NAME": config["key_vault_name"],
+
+        # TODO: why do we need these
```

Comment:
From memory these were defined for the Porter env dict but looks like now you're doing it directly on line 65, so can probably be removed

Reply:
I think to comment is because it doesn't work on Azure (with MSI) without them. I'll check again.

Reply:
We actually need most of those settings on Azure too because they are referenced as credentials in all bundles and also in the credential set. I've updated the comments around those section to make it clearer for the next person.